airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -495,8 +495,9 @@ def _get_ti_pod_labels(context: Context | None = None, include_try_number: bool
         if include_try_number:
             labels.update(try_number=ti.try_number)
         # In the case of sub dags this is just useful
-        if context["dag"].parent_dag:
-            labels["parent_dag_id"] = context["dag"].parent_dag.dag_id
+        if getattr(context["dag"], "parent_dag", False):
+            labels["parent_dag_id"] = context["dag"].parent_dag.dag_id  # type: ignore[attr-defined]
```

Comment:
@kaxil What does this change means for users running newer provider versions with Airflow 2?
Isn't that breaking change? K8s Pod will stop working for SubDAGs

Reply:
from what I see - they wil ljust miss "parent_dag_id" label, as getattr wll handle lack of parent_dag field - and yes, it's a breaking change for those who relied on subdags, but that's expected in this case.

Reply:
but we can avoid it don't we?
we can place this change under check version flag

Reply:
We don't have to. It happens automatically via getattr - it will work in backwards-compatible way if the context will contain parent_dag automatically.

Reply:
But you are quite right - we shouls  (@kaxil?) likely add some flag or at least comment to mark it as removable when we drop Airflow 2 compatibility for providers.

Reply:
>We don't have to. It happens automatically via getattr - it will work in backwards-compatible way if the context will contain parent_dag automatically.

Precisely, this check is backwards-compatible. For someone using sub-dags, it works same as before.

I'll add a TODO in a separate PR to remove this check when we drop Airflow 2 support in Providers.

Reply:
@eladkal Created a PR to address this: https://github.com/apache/airflow/pull/41963